sentry_streams/src/python_operator.rs

```diff
@@ -0,0 +1,520 @@
+                            }
+                            "InvalidMessage" => {
+                                let py_err_obj = err.value(py);
+                                let offset: u64 =
+                                    py_err_obj.getattr("offset").unwrap().extract().unwrap();
+                                let partition = py_err_obj.getattr("partition").unwrap();
+                                Err(SubmitError::InvalidMessage(InvalidMessage {
+                                    offset,
+                                    partition: convert_partition(partition).unwrap(),
+                                }))
+                            }
+                            _ => panic!("Unexpected exception from submit: {}", err),
+                        }
+                    }
+                }
+            })
+        }
+    }
+
+    /// Polls messages from the Python delegate.
+    ///
+    /// This is the method that sends messages to the next ProcessingStrategy.
+    fn poll(&mut self) -> Result<Option<CommitRequest>, StrategyError> {
+        let out_messages = Python::with_gil(|py| -> PyResult<Vec<Py<PyAny>>> {
+            let ret = self.processing_step.call_method0(py, "poll")?;
+            Ok(ret.extract(py).unwrap())
+        });
+
+        match out_messages {
+            Ok(out_messages) => {
+                Python::with_gil(|py| {
+                    self.handle_py_return_value(py, out_messages);
+                });
+                while let Some(msg) = self.transformed_messages.pop_front() {
+                    let commit_request = self.next_strategy.poll()?;
+                    self.commit_request_carried_over = merge_commit_request(
+                        self.commit_request_carried_over.take(),
+                        commit_request,
+                    );
+                    match self.next_strategy.submit(msg) {
+                        Err(SubmitError::MessageRejected(MessageRejected {
+                            message: transformed_message,
+                        })) => {
+                            self.transformed_messages.push_front(transformed_message);
+                            break;
+                        }
+                        Err(SubmitError::InvalidMessage(invalid_message)) => {
+                            return Err(invalid_message.into());
+                        }
+                        Ok(_) => {}
+                    }
+                }
+
+                let commit_request = self.next_strategy.poll()?;
+                Ok(merge_commit_request(
+                    self.commit_request_carried_over.take(),
+                    commit_request,
+                ))
+            }
+            Err(e) => Err(StrategyError::Other(Box::new(e))),
+        }
+    }
+
+    fn terminate(&mut self) {
+        self.next_strategy.terminate();
+    }
+
+    fn join(&mut self, timeout: Option<Duration>) -> Result<Option<CommitRequest>, StrategyError> {
```

Comment:
Simple docstring here could be good to have

this step
- triggers flush (with a timeout) on the delegate, thus returning any final results and clearing any state and closing resources
- submits those results to the next step

Reply:
though, is there a reason why it doesn't call `join` on the next step?

Reply:
> though, is there a reason why it doesn't call join on the next step?\

I think it is a bug. Thanks

Reply:
fixed